microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/model/Property.java

```diff
@@ -170,21 +170,18 @@ public Node findNodeAt(int offset) {
 		if (key == null) {
 			return this;
 		}
-		if (key.getEnd() == -1) {
+		if (key.getEnd() == -1 || offset <= key.getEnd()) {
 			return key;
 		}
 		Node assign = getDelimiterAssign();
 		if (assign == null) {
 			return key;
 		}
-		if (assign.getStart() == offset) {
+		Node value = getValue();
+		if (value == null || offset < value.getStart()) {

```

Comment:
Could you add please add a sample in a comment which correspond to this usecase.